src/Illuminate/Events/CallQueuedListener.php

```diff
@@ -166,7 +166,7 @@ protected function prepareData()
     /**
      * Get the display name for the queued job.
      *
-     * @return string
+     * @return class-string
```

Comment:
I think this change is incorrect, just because the default displayName is the class, name. This could be overwritten and return any string to be displayed. (or am I wrong here?)

Reply:
Could you give me a situation where this would not be a class name?

Reply:
This method currently returns the class string, but if you extend the class than as the method name suggests you can return any "Display name" for underlying queued job.

The Queue already has a mechanism in place which returns a different "displayName" instead of the default class.

https://github.com/laravel/framework/blob/e1db317e7dafa8867b0eada6f89393419ed84cb0/src/Illuminate/Queue/Queue.php#L170-L180

Though this is mainly an internal class, it could also return a different "displayName" instead of the default class (I would never change the current behaviour and I am not sure, many people would do this. But I think a displayName in general is a string and not really a class-string just semantically).

Reply:
I believe the intention of this method is to always show the underlying class, as this is a wrapper class.

We can see examples of Taylor closing PRs related to custom display names: https://github.com/laravel/framework/pull/30551

Although, I can also find an example where the opposite is true: https://github.com/laravel/framework/pull/34620

I'm not going to open a PR to revert this, but I wouldn't be mad if you decided it was best to try a PR to revert it back to `string`.

Reply:
Taylor already reverted it back to string in this PR. So I will leave it as is. But I also completely understand your point why it could be a class-string.

Reply:
Oh, nice. Didn't even realise haha.

Still, was good chatting :)

Reply:
Yeah, for me too ;)